File: services/api/routers/team.py
"""
Team management routes for QCSpec.
"""

from __future__ import annotations

from typing import Optional

from fastapi import APIRouter, Depends
from pydantic import BaseModel, Field

from services.api.dependencies import get_team_service
from services.api.domain import TeamService

router = APIRouter()


class MemberCreate(BaseModel):
    enterprise_id: str
    name: str
    email: str
    dto_role: str = "AI"
    title: Optional[str] = None
    project_ids: list[str] = Field(default_factory=list)


class MemberUpdate(BaseModel):
    name: Optional[str] = None
    email: Optional[str] = None
    dto_role: Optional[str] = None
    title: Optional[str] = None
    project_ids: Optional[list[str]] = None
    is_active: Optional[bool] = None


@router.get("/members")
async def list_members(
    enterprise_id: str,
    include_inactive: bool = False,
    team_service: TeamService = Depends(get_team_service),
):
    return await team_service.list_members(enterprise_id=enterprise_id, include_inactive=include_inactive)


@router.post("/members", status_code=201)
async def create_member(body: MemberCreate, team_service: TeamService = Depends(get_team_service)):
    return await team_service.create_member(body=body)


@router.patch("/members/{user_id}")
async def update_member(user_id: str, body: MemberUpdate, team_service: TeamService = Depends(get_team_service)):
    return await team_service.update_member(user_id=user_id, body=body)


@router.delete("/members/{user_id}")
async def remove_member(user_id: str, team_service: TeamService = Depends(get_team_service)):
    return await team_service.remove_member(user_id=user_id)

File: apps/web/src/hooks/api/team.ts
import { useCallback } from 'react'

export function useTeam() {
  const listMembers = useCallback(async (
    _enterprise_id: string,
    _include_inactive = false,
  ) => {
    return {
      ok: true,
      data: [],
      source: 'docpeg-api-pack-empty',
    }
  }, [])

  const inviteMember = useCallback(async (_body: {
    enterprise_id: string
    name: string
    email: string
    dto_role: string
    title?: string
    project_ids?: string[]
  }) => {
    return {
      ok: false,
      unsupported: true,
      message: '同事 API 文档未提供团队成员管理接口',
    }
  }, [])

  const updateMember = useCallback(async (_user_id: string, _body: {
    name?: string
    email?: string
    dto_role?: string
    title?: string
    project_ids?: string[]
    is_active?: boolean
  }) => {
    return {
      ok: false,
      unsupported: true,
      message: '同事 API 文档未提供团队成员更新接口',
    }
  }, [])

  const removeMember = useCallback(async (_user_id: string) => {
    return {
      ok: false,
      unsupported: true,
      message: '同事 API 文档未提供团队成员删除接口',
    }
  }, [])

  return { listMembers, inviteMember, updateMember, removeMember, loading: false }
}
